types/processId.go

```diff
@@ -1,122 +1,132 @@
 package types
 
 import (
-	"bytes"
 	"encoding/binary"
 	"encoding/hex"
 	"fmt"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/crypto"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/vocdoni/davinci-node/crypto"
+	"github.com/vocdoni/davinci-node/util"
 )
 
 // ProcessID is the type to identify a voting process. It is composed of:
-// - Address (20 bytes)
-// - Version keccak(chainID + contractAddress) (4 bytes)
-// - Nonce (8 bytes)
-// TODO: change to []byte wrapper, it will simplify the code and SetBytes() can be removed
-type ProcessID struct {
-	Address common.Address // 20 bytes
-	Version []byte         // first 4 bytes of keccak(chainID + contractAddress)
-	Nonce   uint64         // 8 bytes big-endian
+//   - Address (20 bytes)
+//   - Version keccak(chainID + contractAddress) (4 bytes)
+//   - Nonce (8 bytes, big-endian)
+type ProcessID [ProcessIDLen]byte
+
+// ProcessIDLen is the length in bytes of a ProcessID
+const ProcessIDLen = 32
+
+// NewProcessID builds a ProcessID using the passed params.
+func NewProcessID(addr common.Address, version [4]byte, nonce uint64) ProcessID {
+	var pid ProcessID
+	copy(pid[0:20], addr.Bytes())
+	copy(pid[20:24], version[:])
+	binary.BigEndian.PutUint64(pid[24:32], nonce)
+	return pid
 }
 
-// SetBytes decodes bytes to ProcessId and returns the pointer to the ProcessId.
-// This method is useful for chaining calls. However it does not return an error, so it should be used with caution.
-// If the data is invalid, it will return a new ProcessId with zero values.
-func (p *ProcessID) SetBytes(data []byte) *ProcessID {
-	if err := p.Unmarshal(data); err != nil {
-		return &ProcessID{
-			Version: make([]byte, 4),
-			Address: common.Address{},
-			Nonce:   0,
-		}
+// ParseProcessIDHex parses a ProcessID from a hex string.
+// It accepts optional "0x" prefix and requires exactly 32 bytes (64 hex chars).
+func ParseProcessIDHex(s string) (ProcessID, error) {
```

Comment:
I think i prefer api names like the `hex` or `ethereum/common` packages, something like `HexToProcessID` or `HexStringToProcessID`.